personalized_nlp/learning/regressor.py

```diff
@@ -45,6 +45,26 @@ def training_step(self, batch, batch_idx):
     def training_epoch_end(self, outs):
         pass
 
+    def freeze(self) -> None:
+        for name, param in self.named_parameters():
+            if 'fc' not in name:
+                param.requires_grad = False
+        self.model.frozen = True
```

Comment:
You should verify if `model` object has the parameter `frozen`. `Regressor` is a class which can work with `model` objets which are of vary different classes, not necesarilly `UserID` one.

The same should be done in `unfreeze` methods and consequently in `Classifier` class.